### PR TITLE
Clean up CMakeLists.txt

### PR DESCRIPTION
**Pull request by gsjaardema**
A few modifications to hopefully make configuration easier.
* Add CMP0074 policy which eliminates potentially confusing warning about ignoring `HDF5_ROOT` if it is set.   The variable *is* used in the legacy `find_package` call, so issuing a warning saying it is ignored could cause user to undefined that variable and change the results of the legacy `find_package` (Voice of experience...)
* The `set` calls that were deleted (lines 234 to 238) are repeated later in the file so are redundant.
* I don't see anywhere that `HDF5_LIBRARY_DIRS` is set; I think it should be `HDF5_LIBRARY`

It looks like `HDF5_DIR` is used if the HDF5 build/installation used CMake and `HDF5_ROOT` is used if HDF5 used a configure-based build.

### CODE REVIEW

**Comment by gsjaardema** [comment]
I will stop making changes now.  Sorry for the churn.

File: CMakeLists.txt [hunk 1]
```diff
@@ -200,6 +200,15 @@ if (CGNS_ENABLE_HDF5)
...
   set (SEARCH_PACKAGE_NAME "hdf5")
+  
+  # If the first `find_package` below does not succeed, then the legacy `find_package`
+  # is tried (the `else` below).  The legacy find_package uses `HDF5_ROOT`.  But if 
+  # this is set, then CMake will issue warning and mistakenly say that `HDF5_ROOT` is
+  # not used even though it might be.  This can confuse user, so set policy to not
+  # issue that warning.
+  if (${CMAKE_VERSION} GREATER "3.13")
+     cmake_policy(SET CMP0074 NEW)
+  endif()
```
[the remark this answers is not in the document]
Comment (gsjaardema) [reply CMakeLists.txt]: Sorry, didn't see this until now.  Thanks for making the changes.

[the remark this answers is not in the document]
Comment (gsjaardema) [reply CMakeLists.txt]: Do you want me to add the CMP0081 or are you doing it?

**Comment by brtnfld** [comment]
I'll run these changes through our Daily testing this weekend.

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (brtnfld) [reply CMakeLists.txt]: I'll add it.

**Comment by brtnfld** [comment]
The current branch fails for me when building with a cmake build of HDF5,

-- HDF5 find comps: C;static
-- HDF5 C libs:1 static:1 and shared:
-- HDF5-1.13.0 found: INC= TOOLS=/home/brtnfld/packages/hdf5/build/hdf5/bin HDF5_BUILD_SHARED_LIBS=YES
-- HDF5 link libs: hdf5::hdf5-static

[  6%] Building C object src/CMakeFiles/cgns_static.dir/cgio_ftoc.c.o
cd /home/brtnfld/packages/CGNS/build/src && /home/brtnfld/packages/mpich/bin/mpicc  -DLOWERCASE_ -I/home/brtnfld/packages/CGNS/build/src -I/home/brtnfld/packages/CGNS/src  -g -g   -o CMakeFiles/cgns_static.dir/cgio_ftoc.c.o   -c /home/brtnfld/packages/CGNS/src/cgio_ftoc.c
/home/brtnfld/packages/CGNS/src/adfh/ADFH.c:42:10: fatal error: hdf5.h: No such file or directory
 #include "hdf5.h"
          ^~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/cgns_static.dir/build.make:141: src/CMakeFiles/cgns_static.dir/adfh/ADFH.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/brtnfld/packages/CGNS/src/cgns_internals.c:38:10: fatal error: hdf5.h: No such file or directory
 #include "hdf5.h"
          ^~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/cgns_static.dir/build.make:76: src/CMakeFiles/cgns_static.dir/cgns_internals.c.o] Error 1
/home/brtnfld/packages/CGNS/src/cgnslib.c:57:11: fatal error: hdf5.h: No such file or directory
 # include "hdf5.h"
           ^~~~~~~~
compilation terminated.
make[2]: *** [src/CMakeFiles/cgns_static.dir/build.make:102: src/CMakeFiles/cgns_static.dir/cgnslib.c.o] Error 1
/home/brtnfld/packages/CGNS/src/pcgnslib.c:30:10: fatal error: hdf5.h: No such file or directory
 #include "hdf5.h"

My build:

cmake \
    -D CMAKE_C_COMPILER:PATH="$MPI/mpicc" \
    -D CMAKE_C_FLAGS:STRING="-g" \
    -D CMAKE_Fortran_FLAGS:STRING="-g" \
    -D CMAKE_BUILD_TYPE:STRING="Debug" \
    -D CMAKE_Fortran_COMPILER:PATH="$MPI/mpif90" \
    -D CGNS_BUILD_SHARED:BOOL=OFF \
    -D CGNS_USE_SHARED:BOOL=OFF \
    -D CMAKE_STATIC_LINKER_FLAGS:STRING="" \
    -D CGNS_ENABLE_TESTS:BOOL=ON \
    -D CGNS_BUILD_CGNSTOOLS:BOOL=ON \
    -D CGNS_ENABLE_SCOPING:BOOL=OFF \
    -D CMAKE_INSTALL_PREFIX:PATH="./" \
    -D CMAKE_EXE_LINKER_FLAGS:STRING="" \
    -D CMAKE_PREFIX_PATH=$HDF5 \
    -D CGNS_ENABLE_PARALLEL:BOOL=ON \
    -D HDF5_NEED_MPI:BOOL=ON \
    -D CGNS_ENABLE_LFS:BOOL=OFF \
    -D CGNS_ENABLE_HDF5:BOOL=ON \
    -D HDF5_NEED_ZLIB:BOOL=ON \
    -D HDF5_NEED_SZIP:BOOL=OFF \
    -D CGNS_ENABLE_FORTRAN:BOOL=ON \
    -D CGNS_ENABLE_64BIT:BOOL=ON \
    $CGNS

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,11 +3,15 @@ if(COMMAND cmake_policy)
...
+  cmake_policy(SET CMP0015 NEW)
+endif()
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.13")
+  cmake_policy(SET CMP0081 NEW)
+endif()
+
 set(CMAKE_BUILD_TYPE "Release" CACHE STRING
   "one of: Release, Debug, RelWithDebInfo or MinSizeRel")
-if (${CMAKE_MAJOR_VERSION} GREATER 2 OR ${CMAKE_MINOR_VERSION} GREATER 7)
-  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Release;Debug;RelWithDebInfo;MinSizeRel")
```
Comment (MicK7) [inline CMakeLists.txt]: Why is this line removed ?

Comment (gsjaardema) [reply CMakeLists.txt]: I removed the `if` test since the minimum required CMake version is 2.8 so any CMake that makes it this far would meet the conditions of the if test.

I removed the `set_property` since I don't think that it is needed for current versions of CMake.  The only difference that I can see in the CMakeCache.txt with and without this line is an entry in the "INTERNAL Cache Entries" portion of the file.  The `CMAKE_BUILD_TYPE` variable up in the "EXTERNAL Cache Entries" portion of the file is the same with or without the `set_property`

Comment (MicK7) [reply CMakeLists.txt]: Ok thanks for the explanation

File: CMakeLists.txt [hunk 3]
```diff
@@ -257,12 +264,14 @@ if (CGNS_ENABLE_HDF5)
   set (HDF5_PACKAGE_NAME ${SEARCH_PACKAGE_NAME})
 
   if (HDF5_FOUND)
+    if (WIN32)
```
Comment (MicK7) [inline CMakeLists.txt]:
```suggestion
    if (NOT DEFINED HDF5_INCLUDE_DIRS)
```

Comment (gsjaardema) [reply CMakeLists.txt]: Yes, that is probably a better option.  It looks like some `FindHDF5.cmake` packages are setting `HDF5_INCLUDE_DIR` instead of `HDF5_INCLUDE_DIRS`.  Based on the CMake documentation, `HDF5_INCLUDE_DIR` was deprecated as of CMake-2.8.1, but it looks like the build is still using the older versions in some cases.
